src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSASignatureDeformatter.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Security.Cryptography
+{
+    public class DSASignatureDeformatter : AsymmetricSignatureDeformatter
+    {
+        private DSA _dsaKey;
+
+        public DSASignatureDeformatter() { }
+
+        public DSASignatureDeformatter(AsymmetricAlgorithm key) : this()
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            _dsaKey = (DSA)key;
+        }
+
+        public override void SetKey(AsymmetricAlgorithm key)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            _dsaKey = (DSA)key;
+        }
+
+        public override void SetHashAlgorithm(string strName)
+        {
+            try
+            {
+                // Verify the name is a valid HashAlgorithm even though it is not currently used
+                Oid.FromFriendlyName(strName, OidGroup.HashAlgorithm);
```

Comment:
Per discussion, hard-code to SHA1 for now (to match desktop) and then log an issue to allow additional